userspace/falco/app/actions/configure_interesting_sets.cpp

```diff
@@ -0,0 +1,195 @@
+/*
+Copyright (C) 2023 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "actions.h"
+
+using namespace falco::app;
+using namespace falco::app::actions;
+
+static void extract_base_syscalls_names(
+		const std::unordered_set<std::string>& base_syscalls_names,
+		std::unordered_set<std::string>& user_positive_names,
+		std::unordered_set<std::string>& user_negative_names)
+{
+	for (const std::string &ev : base_syscalls_names)
+	{
+		if (!ev.empty())
+		{
+			if (ev.at(0) == '!')
+			{
+				user_negative_names.insert(ev.substr(1, ev.size()));
+			}
+			else
+			{
+				user_positive_names.insert(ev);
+			}
+		}
+	}
+}
+
+static void check_for_rules_unsupported_events(falco::app::state& s, const libsinsp::events::set<ppm_sc_code>& rules_sc_set)
+{
+	/* Unsupported events are those events that are used in the rules
+	 * but that are not part of the selected event set. For now, this
+	 * is expected to happen only for high volume I/O syscalls for
+	 * performance reasons. */
+	auto unsupported_sc_set = rules_sc_set.diff(s.selected_sc_set);
+	if (unsupported_sc_set.empty())
+	{
+		return;
+	}
+
+	/* Get the names of the events (syscall and non syscall events) that were not activated and print them. */
+	auto names = libsinsp::events::sc_set_to_names(unsupported_sc_set);
+	std::cerr << "Loaded rules match event types that are not activated or unsupported with current configuration: warning (unsupported-evttype): " + concat_set_in_order(names) << std::endl;
+	std::cerr << "If syscalls in rules include high volume I/O syscalls (-> activate via `-A` flag), else syscalls might be associated with syscalls undefined on your architecture (https://marcin.juszkiewicz.com.pl/download/tables/syscalls.html)" << std::endl;
```

Comment:
... ", else syscalls may have been removed via base_syscalls option or might be associated ..."